display_list/geometry/dl_region.cc

```diff
@@ -711,15 +709,35 @@ bool DlRegion::intersects(const DlRegion& region) const {
   }
 
   auto ours = lines_.begin();
+  auto ours_end = lines_.end();
   auto theirs = region.lines_.begin();
+  auto theirs_end = region.lines_.end();
+
+  FML_DCHECK(ours != ours_end && theirs != theirs_end);
+
+  auto ours_len = ours_end - ours;
+  auto their_len = theirs_end - theirs;
+
+  if (ours_len > kBinarySearchThreshold &&
```

Comment:
This code is repeated often enough - should it be put in a small inlineable static function or method?

Reply:
Also, could intersects-single-rect benefit from the same threshold logic?